### PR TITLE
Implemented some CityObjectElements and enabled tesselate

### PR DESCRIPTION
**Pull request by HarutakaMatsumoto**
Hello!
I updated the following points of libcitygml.
1. Enabled tesselate. If you set tesselate is false, tessellation is skipped.
2. Implemented RectifiedGridCoverage.
3. Implemented ExternalReference.
4. Implemented Lod0Geometry
5. Implemented Loc0ImplicitRepresentation
6. Implemented Lod0TerrainIntersection
7. Implemented Lod0Network
8. Implemented Lod0MultiCurve
9. Implemented Lod0MultiSurface
10. Implemented GeneralizesTo

This commit successes parsing [533925_tran_6697_op2.gml](https://gic-plateau.s3-ap-northeast-1.amazonaws.com/2020/tokyo23ku/533925_2.zip).
Please check and merge this.

### CODE REVIEW

File: sources/include/citygml/citygml.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ namespace citygml
             , minLOD( 0 )
             , maxLOD( 4 )
             , optimize( false )
```
Comment (jklimke) [inline sources/include/citygml/citygml.h:71]: as the current behavior was to run the tessellation by default, an initialization of the flag to false may break things for user. default initialization to true might be better

File: sources/include/citygml/citygmlfactory.h [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 #include <citygml/geometry.h>
 #include <citygml/cityobject.h>
+#include "citygml/externalreference.h"
```
Comment (jklimke) [inline sources/include/citygml/citygmlfactory.h:5]: please use the same import style <> as before

File: sources/include/citygml/cityobject.h [hunk 3]
```diff
@@ -6,6 +6,8 @@
 #include <citygml/featureobject.h>
 #include <citygml/citygml_api.h>
 #include <citygml/enum_type_bitmask.h>
+#include "citygml/rectifiedgridcoverage.hpp"
```
Comment (jklimke) [inline sources/include/citygml/cityobject.h:9]: please adhere to current import style <>

File: sources/src/citygml/externalreference.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+//
+//  externalreference.cpp
+//  libcitygml
+//
+//  Created by 松本青空 on 2021/11/15.
+//
+
+#include "citygml/externalreference.h"
```
Comment (jklimke) [inline sources/src/citygml/externalreference.cpp:8]: import style

**Review by jklimke — CHANGES_REQUESTED** [review]
First of all: Thank you for you contribution.

Please have a look at the annotations i did, please adapt the import style. As far as i remember public headers, i.e., the ones starting with 'citygml' should be imported using import <header_name>

Changing the default tesselation to false may break other applications that use the library as the behaviour changes.

File: sources/src/citygml/externalreference.cpp (hunk 4 above)
Comment (HarutakaMatsumoto) [reply sources/src/citygml/externalreference.cpp:8]: Do you mean the following fix?
```suggestion
#import "citygml/externalreference.h"
```

Comment (jklimke) [reply sources/src/citygml/externalreference.cpp:8]: no ... even easier  #include <citygml/externalreference.h>

Comment (HarutakaMatsumoto) [reply sources/src/citygml/externalreference.cpp:8]: OK.

**Comment by HarutakaMatsumoto** [comment]
@jklimke 
Fixed for annotations, and aligned the format.
Please check.

**Comment by HarutakaMatsumoto** [comment]
Thank you for your checking and merging!

**Pull request merged**